static/js/subs.js

```diff
@@ -698,14 +702,7 @@ $(function () {
             return;
         }
```

Comment:
Overall this change looks great, except that it breaks the unusual case where `should_list_all_streams` is false -- basically it becomes impossible to subscribe to a stream because there's a "Subscibe" button with no text box to enter the stream name.  

`should_list_all_streams` being false is for a somewhat hacky scenario related to mirroring of MIT Zephyr, where following the convention in Zephyr itself, all streams are "public" in the sense that anyone can join, yet the stream names are secret so you need to know the name of a stream to subscribe.  The stream creation modal is never used for Zephyr mirroring because there aren't any stream-level settings one can set there.  The details of how that all works are somewhat complex, but I think we need to do in order to avoid breaking !should_list_all_streams is to modify the template and CSS to have the old page shape still present if !should_list_all_streams.  Should be easy to do in the template -- just pass in the return value of should_list_all_streams() as an argument, and use that to control whether the button appears on that page (and what CSS it has on it).  

You can test your work pretty easily by patching the `should_list_all_streams` function to return false temporarily.

Thanks for working on this @disissid!